### ISSUE TITLE
Client has problems receiving too many packets

### ISSUE THREAD

**Issue by lRaikeNl** [Eastrall/Ether.Network#98]
Hi there. The server side is processing just fine with multiple incoming packets, just the client stops receiving packets at some point when I try to "spam" the client with packets from the server.

I tried to debug it:

NetServer.cs:
```cs
...
       private void SendMessage(MessageData messageData)
        {
            SocketAsyncEventArgs writeEventArgs = this._writePool.Pop();

            if (writeEventArgs != null)
            {
                writeEventArgs.SetBuffer(messageData.Message, 0, messageData.Message.Length);
                writeEventArgs.UserToken = messageData.User;

                if (!messageData.User.Socket.SendAsync(writeEventArgs)) // this line processes just fine
                    this.ProcessSend(writeEventArgs);
            }
            else
            {
                this._autoSendEvent.WaitOne();
                this.SendMessage(messageData);
            }
        }
```

NetClient.cs:
```cs
public void Connect()
...  

            this._sendingQueueWorker.Start();
            this._receivingQueueWorker.Start();
            this.Token.Socket = this.Socket;
            this.Token.MessageHandler = data => // won't jump inside this method after a couple of tries
            {
                try
                {
                    this._receivingQueue.Add(data, this._cancelToken);
                }
                catch (OperationCanceledException)
                {
                    this.Disconnect();
                }
            };
            this.IsRunning = true;

            if (!this.Socket.ReceiveAsync(this._socketReceiveArgs))
                this.ProcessReceive(this._socketReceiveArgs);
        }
```

**Comment by lRaikeNl** [comment]
I think it has something to do with the blocking queue, but not sure though.

**Comment by lRaikeNl** [comment]
Only appears if multiple packets are sent to the client.

**Comment by Eastrall** [comment]
Hello, thank you for reporting this issue.
This might be related with issue #90 about a code duplication. A PR will be done soon to fix this issue. 😄

**Comment by lRaikeNl** [comment]
Case can be closed. I tried to reproduce the bug but it seems to be fixed. Thank you very much!!

**Comment by Altiano** [comment]
I think I have the same problem right now.

In my case, I need to send 20 MB file to ~20 client at the same time.

I try to break down the file to 1 MB each so that the Idea is that I only require 
1 MB * 20 (clients) = 20 MB byte array memory at a time, instead of 20 * 20 (clients) = 400 MB

If the client and the server on the same machine (In development environment), 
my experiment works, but on actual network, it doesn't

Any Idea?

            For chunkIndex As Integer = 1 To maxLoop
              Dim skip As Integer = chunkIndex * CHUNK_SIZE

              Dim chunkResponse As Byte() = byteResponse.Skip(skip).Take(CHUNK_SIZE).ToArray()

              Using chunkPacketResponse As New NetPacket
                chunkPacketResponse.Write(Of Byte())(chunkResponse)
                Me.Send(chunkPacketResponse)
              End Using
            Next

That's the server, CHUNK_SIZE is 1 MB,

On client, `HandleMessage()`, only the first chunk is received.

**Comment by Eastrall** [comment]
This issue has been solved for version `3.0.1` and the fix is not shipped in version `3.0.0`. I'll try to make a package as soon has I can that will correct some minor issues.

**Comment by Altiano** [comment]
Whew got me hours 😄 

Anyway, good job bro. 👍
